packages/odyssey-react/src/components/Modal/index.tsx

```diff
@@ -0,0 +1,154 @@
+}
+
+export type PropsModalFooter = { 
+  children: ReactNode
+}
+
+export type PropsModalButton = { 
+  close?: boolean,
+  children: ReactText,
+  variant?: ButtonVariants,
+  onClick?: () => void
+}
+
+export type StaticComponents = {
+  Header: FunctionComponent<PropsModalHeader>,
+  Body: FunctionComponent<PropsModalBody>,
+  Footer: FunctionComponent<PropsModalFooter>,
+  Button: FunctionComponent<PropsModalButton>
+}
+
+/**
+ * UI that appears on top of the main content and moves the system into a mode 
+ * requiring user interaction. This dialog disables the main content until the 
+ * user interacts with the modal dialog.
+ * 
+ * @todo OKTA-00000: [React] Modal: Use proper close icon
+ * @todo Determine how to enforce use of only Modal static components as children using TS
+ * 
+ * @component
+ * @example
+ * <Modal open={true} onOpen={()=>{}} onClose={()=>{}}>
+ *  <Modal.Header>Modal Title</Modal.Header>
+ *  <Modal.Body>
+ *    <p>This is the modal content area. It's width is determined based on the amount of content within it.</p>
+ *  </Modal.Body>
+ *  <Modal.Footer>
+ *    <Button variant="clear" close>Cancel</Button>
+ *    <Button onClick={() => {console.log('do something')}}Continue</Button>
+ *  </Modal.Footer>
+ * </Modal>
+ */
+
+export const ModalContext = createContext<{ onClose: () => void}>({ onClose: () => void 0 });
+
+const Modal: FunctionComponent<PropsModal> & StaticComponents = (props) => {
+  const { children, id, open = false, onClose, onOpen } = props
+  const context = useMemo(() => ({ onClose, onOpen }), [onClose, onOpen]);
+  const oid = useOid(id);
+
+  const componentClass = useCx(
+    'ods-modal',
+    { "is-open": open }
+  );
+
+  if (open && onOpen) {
+    onOpen();
+  }
+
+  return createPortal(
+    <ModalContext.Provider value={context}>
+      <div className={componentClass} id={oid} aria-hidden={!open}>
+        <div className="ods-modal--overlay" tabIndex={-1}>
+          <div className="ods-modal--dialog" role="dialog" aria-modal="true" aria-labelledby="ods-modal-standard-title">
+            {children}
+          </div>
+        </div>
+      </div>
+    </ModalContext.Provider>,    
+    document.body
+  )
+};
+
+Modal.Header = ({ children }) => (
+  <header className="ods-modal--header">
+    <span className="ods-modal--dismiss">
+      <Modal.Button close variant="dismiss">&#x2716;</Modal.Button>
+    </span>
+    <h1 className="ods-modal--title" id="ods-modal-standard-title">
+      {children}
+    </h1>
+  </header>
+);
+
+Modal.Body = ({ children }) => (
+  <main className="ods-modal--content" id="ods-modal-standard-content">
+    {children}
+  </main>
+);
+
+Modal.Footer = ({ children }) => (
+  <footer className="ods-modal--footer">
+    {children}
+  </footer>
+);
+
+Modal.Button = ({ children, variant, close, onClick }) => {
+  const { onClose } = useContext(ModalContext);
+
+  return <Button variant={variant} onClick={close ? () => onClose() : onClick}>{children}</Button>
+};
```

Comment:
+1  I love that you re-used Button here but wrapped it in another Modal focused API